### The three questions to be answered for this assignment:

1. What is the 
northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

In [1]:
#import pandas and numpy
import pandas as pd
import numpy as np
import matplotlib
from ipywidgets import *
import os
from ipyleaflet import * #imports ipyleaflet to check location that shows the coordinates are erroneous.

In [2]:
#imports the url and creates a pandas dataframe df
url="https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv"
df = pd.read_csv(url)
df.shape #verifies that all data from source was captured in the dataframe. 

(1458, 8)

## Question 1 
**Assumption:** The point with the highest positive lattitude value is the northmost location.

**Approach:** To get the northmost location I will sort the dataframe in descending order by the 'lat' column. Then the top will be obtained by displaying the first 5 row in the sorted dataframe. 


In [3]:
northmost = df.sort_values('lat', ascending=False).head(5)
northmost

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt   tz dst              tzone  
417  149 -5.0   A                 \N  
230   44 -9.0   A  America/Anchorage  
110   41 -9.0   A  America/Anchorage  
708   35 -9.0   A  America/Anchorage  
152   96 -9.0   A  America/Anchorage

**Answer:** The airport with the northmost location is Dilland Hopkins Airport with a lattitude of 72.270833. However this answer is incorrect because the coordinates are in the middle of the Barents Sea as show below in the map. The second row from the above table is correct; this was verified again with mapping the coordinates which are in Anchorage as stated in the "tzone" column for index 230.

In [22]:
center = 72.270833, 42.898333
m = Map(center=(center), basemap=basemaps.Esri.WorldStreetMap, marker=True, zoom=3)
marker = Marker(location=center)
m.add_layer(marker)
m

Map(basemap={'url': 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{…

In [21]:
center = 71.285446, -156.766003
m2 = Map(center=(center), basemap=basemaps.Esri.WorldStreetMap, marker=True, zoom=3)
marker = Marker(location=center)
m2.add_layer(marker)
m2

Map(basemap={'url': 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{…

## Question 2
**Assumption:** The point with the highest positive longitude value is the easternmost location.

**Approach:** To get the easternmost location I will sort the dataframe in descending order by the 'lon' column. Then the top will be obtained by displaying the first 5 rows in the sorted dataframe. 

In [6]:
easternmost = df.sort_values('lon', ascending=False).head(5)
easternmost

faa                           name        lat         lon   alt   tz  \
1290  SYA                   Eareckson As  52.712275  174.113620    98 -9.0   
942   MYF               Montgomery Field  32.475900  117.759000    17  8.0   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478  8.0   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149 -5.0   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45 -5.0   

     dst              tzone  
1290   A  America/Anchorage  
942    A     Asia/Chongqing  
396    A     Asia/Chongqing  
417    A                 \N  
444    A   America/New_York

**Answer:** The easternmost airport is Eareckson As with a longitude of 174.11362. This is correct coordinates according to Wikipedia which places the airport on Shemya Island, Alaska. The map coordinates below confirm the correct location. 

In [7]:
center = 52.712275, 174.113620
m2 = Map(center=(center), basemap=basemaps.Esri.WorldStreetMap, marker=True, zoom=13)
marker = Marker(location=center)
m2.add_layer(marker)
m2

Map(basemap={'url': 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{…

## Question 3
**Assumptions:** The data will come from the table with weather information. I will need to import another data table and query the information from that table.

**Approach:** To get the data table from Github I will use the same approach as for the first two questions. Then a query will be used to pull the data for the date, Feb 12th, 2013, and max will be used for the column with wind speeds.

In [8]:
#imports csv from Github and creates a new dataframe -- checks by first 5 rows.
url2 = "https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv"
df2 = pd.read_csv(url2)
df2.shape

(26115, 15)

In [9]:
df2_date = df2[(df2.year == 2013) & (df2.month == 2) & (df2.day == 12)]
df2_date.sort_values('wind_speed', ascending=False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z

**Answer:** There appears to be a data entry error. The wind speed list for origin EWR on Feb 12, 2013 is 1,048 which is an error as there has never been wind speeds recorded of this magnitude on earth. The correct answer is the next greatest value of 23.01560 recorded at LGA.